Extension/src/LanguageServer/client.ts

```diff
@@ -3506,12 +3510,40 @@ export class DefaultClient implements Client {
                     character: range.end.character,
                     line: range.end.line
                 }
-            }
+            },
+            copyToClipboard: copy ?? false
         };
 
         const result: CreateDeclarationOrDefinitionResult = await this.languageClient.sendRequest(CreateDeclarationOrDefinitionRequest, params);
-        // TODO: return specific errors info in result.
-        if (result.changes === undefined) {
+        // Create/Copy returned no result.
+        if (result.edit === undefined) {
+            await vscode.window.showInformationMessage(result.errorText); // Copy/Create Declaration/Definition was completely unsuccessful due to api failure.
+            return;
+        }
+
+        // Handle CDD error messaging
+        if (result.errorText) {
+            try {
+                await vscode.env.clipboard.writeText(result.clipboardText);
+                await vscode.window.showInformationMessage(result.errorText + " Declaration/definition was copied to clipboard."); // CDD failed but fallback to clipboard was succesful
+
+            } catch {
+                // handle failure
+                await vscode.window.showInformationMessage(result.errorText); // CDD failed but fallback was unsuccessful.
```

Comment:
Never await a `showInformationMessage` (or other vscode functions that display UI).  This will cause the code here to pause until the user dismisses the UI (which could be never).  You can omit the await here.

Reply:
To clarify, it's not always bad to await these, such as when you need to know which of multiple options the user clicked on, and provided that occurs within an async that is not being awaited on at some higher level.  But, awaiting here will stall the async execution of this particular async context, and that's probably not what you want.